### PR TITLE
Add Trino 458 release notes

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 458 release.

## Verification for each pull request

Format: PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 7 Sep 2024

* #23329 ✅ rn ✅ docs

## 8 Sep 2024

* #23332 ✅ rn ✅ docs
* #23335 ✅ rn ✅ docs

## 9 Sep 2024

* #23312 ✅ rn ✅ docs
* #23337 ✅ rn ✅ docs
* #23336 ✅ rn ✅ docs
* #23330 ✅ rn ✅ docs
* #23340 ✅ rn ✅ docs
* #23158 ✅ rn ✅ docs
* #23341 ✅ rn ✅ docs
* #23286 ✅ rn ✅ docs
* #23325 ✅ rn ✅ docs
* #23331 ✅ rn ✅ docs

## 10 Sep 2024

* #23320 ✅ rn ✅ docs
* #23326 ✅ rn ✅ docs
* #23351 ✅ rn ✅ docs
* #23087 ✅ rn ✅ docs
* #23323 ✅ rn ✅ docs
* #23353 ✅ rn ✅ docs

## 11 Sep 2024

* #23342 ✅ rn ✅ docs
* #23327 ✅ rn ✅ docs
* #23358 ✅ rn ✅ docs
* #23360 ✅ rn ✅ docs
* #23362 ✅ rn ✅ docs
* #23364 ✅ rn ✅ docs
* #23355 ✅ rn ✅ docs
* #23368 ✅ rn ✅ docs
* #23350 ✅ rn ✅ docs

## 12 Sep 2024

* #23253 ✅ rn ✅ docs
* #23052 ✅ rn ✅ docs
* #23375 ✅ rn ✅ docs
* #23385 ✅ rn ✅ docs
* #23381 ✅ rn ✅ docs
* #23365 ✅ rn ✅ docs
* #23388 ✅ rn ✅ docs
* #23377 ✅ rn ✅ docs
* #12802 ✅ rn ✅ docs
* #23370 ✅ rn ✅ docs
* #23371 ✅ rn ✅ docs
* #23380 ✅ rn ✅ docs

## 13 Sep 2024

* #23395 ✅ rn ✅ docs
* #23239 ✅ rn ✅ docs
* #23378 ✅ rn ✅ docs
* #23397 ✅ rn ✅ docs
* #23339 ✅ rn ✅ docs
* #23398 ✅ rn ✅ docs
* #23346 ✅ rn ✅ docs
* #23304 ✅ rn ✅ docs
* #23409 ✅ rn ✅ docs
* #23391 ✅ rn ✅ docs
* #23321 ✅ rn ✅ docs
* #23392 ✅ rn ✅ docs
* #23411 ✅ rn ✅ docs
* #23343 ✅ rn ✅ docs
* #23366 ✅ rn ✅ docs
* #23413 ✅ rn ✅ docs
* #23415 ✅ rn ✅ docs

## 14 Sep 2024

* #23420 ✅ rn ✅ docs
* #23421 ✅ rn ✅ docs
* #23412 ✅ rn ✅ docs
* #23423 ✅ rn ✅ docs

## 15 Sep 2024

* #23430 ✅ rn ✅ docs
* #23424 ✅ rn ✅ docs
* #23422 ✅ rn ✅ docs

## 16 Sep 2024

* #23431 ✅ rn ✅ docs
* #23436 ✅ rn ✅ docs
* #23444 ✅ rn ✅ docs
* #23440 ✅ rn ✅ docs
* #23435 ✅ rn ✅ docs
* #23449 ✅ rn ✅ docs
* #23452 ✅ rn ✅ docs
* #23401 ✅ rn ✅ docs
* #23448 ✅ rn ✅ docs

## 17 Sep 2024

* #21979 ✅ rn ✅ docs
* #23461 ✅ rn ✅ docs
* #23458 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-458.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Release 458 (11 Sep 2024)
+
+## General
+
+* Improve performance for queries with a redundant `DISTINCT` clause on an
```
Comment (martint) [inline docs/src/main/sphinx/release/release-458.md:5]: I would remove the "on an already-distinct ..." part. It's confusing and doesn't add much value to describe that.

File: docs/src/main/sphinx/release/release-458.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+# Release 458 (13 Sep 2024)
+
+## General
+
+* Improve performance for queries with a redundant `DISTINCT` clause. ({issue}`23087`)
+* Improve error message while dereferencing an unknown row field. ({issue}`23370`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-458.md:6]: We don’t need to document improvements to error messages.

File: docs/src/main/sphinx/release/release-458.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+## JDBC
+
+* Reduce data transfer from remote systems for queries with large IN lists. ({issue}`23381`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-458.md:10]: Is this for jdbc driver? I’m confused.

Comment (mosabua) [reply docs/src/main/sphinx/release/release-458.md]: updated

File: docs/src/main/sphinx/release/release-458.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-458.md]: nixed

File: docs/src/main/sphinx/release/release-458.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  `fs.hadoop.enabled` in each catalog. Use the migration guides for [Azure
+  Storage](fs-legacy-azure-migration), [Google Cloud
+  Storage](fs-legacy-gcs-migration), and [S3](fs-legacy-s3-migration) to assist
+  if you have not switched from legacy support. ({issue}`23343`)
+* Add JMX monitoring to the native S3 file system support. ({issue}`23177`)
+* Reduce the number of file system operations when reading tables with file system
+  caching enabled. ({issue}`23327`)
+* Improve the `flush_metadata_cache` procedure to include flushing the file
+  status cache. ({issue}`22412`)
+* Fix failure when listing Hive tables from Glue having unsupported syntax. ({issue}`23253`)
```
Comment (ebyhr) [inline docs/src/main/sphinx/release/release-458.md:44]: I don't understand what "unsupported syntax" means in this context. 

cc: @pajaks

File: docs/src/main/sphinx/release/release-458.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+* {{breaking}} Deactivate [legacy file system support](file-system-legacy) for
+  all catalogs. You must activate the desired [file system
+  support](file-system-configuration) with
+  `fs.native-azure.enabled`,`fs.native-gcs.enabled`, `fs.native-s3.enabled`, or
+  `fs.hadoop.enabled` in each catalog. Use the migration guides for [Azure
+  Storage](fs-legacy-azure-migration), [Google Cloud
+  Storage](fs-legacy-gcs-migration), and [S3](fs-legacy-s3-migration) to assist
+  if you have not switched from legacy support. ({issue}`23343`)
+* Add JMX monitoring to the native S3 file system support. ({issue}`23177`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-458.md:21]: "support" at the end of the sentence sounds weird. Rephrase either as "Add support for JMX monitoring to ...", or simply remove the word "support" at the end.

File: docs/src/main/sphinx/release/release-458.md (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-458.md:48]: but what is a Hive unsupported table

File: docs/src/main/sphinx/release/release-458.md (hunk 5 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-458.md:25]: Changed to link

File: docs/src/main/sphinx/release/release-458.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+## Redshift connector
+
+* Reduce data transfer from remote systems for queries with large `IN` lists. ({issue}`23381`)
+
+## SingleStore connector
+
+* Reduce data transfer from remote systems for queries with large `IN` lists. ({issue}`23381`)
+
+## Snowflake connector
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-458.md:98]: @raunaqmorarka this seems wrong .. a least in the docs there is nothinfg about domain compaction.

File: docs/src/main/sphinx/release/release-458.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
+# Release 458 (16 Sep 2024)
+
+## General
+
+* Improve performance for queries with a redundant `DISTINCT` clause. ({issue}`23087`)
+
+## ClickHouse connector
+
+* Reduce data transfer from remote systems for queries with large `IN` lists. ({issue}`23381`)
```
Comment (raunaqmorarka) [inline docs/src/main/sphinx/release/release-458.md:9]: Please skip this, the default for ClickHouse was already customized to a higher value before

File: docs/src/main/sphinx/release/release-458.md (hunk 6 above)
Comment (raunaqmorarka) [reply docs/src/main/sphinx/release/release-458.md:98]: SF connector is extending JdbcPlugin, so it gets this functionality by default

File: docs/src/main/sphinx/release/release-458.md (hunk 4 above)
Comment (pajaks) [reply docs/src/main/sphinx/release/release-458.md]: > but what is a Hive unsupported table

It's table which cannot be converted to Trino table and throws for example: `HIVE_UNSUPPORTED_FORMAT`

File: docs/src/main/sphinx/release/release-458.md (hunk 6 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-458.md:98]: https://github.com/trinodb/trino/pull/23463

File: docs/src/main/sphinx/release/release-458.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
+# Release 458 (17 Sep 2024)
+
+## General
+
+* Improve performance for queries with a redundant `DISTINCT` clause. ({issue}`23087`)
+
+ ## JDBC
+
+* Add support for monitoring with OpenTelemetry. ({issue}`23458`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-458.md:9]: fyi @wendigo

Comment (wendigo) [reply docs/src/main/sphinx/release/release-458.md:9]: I'd phrase that differently. Add support for tracing JDBC driver with OpenTelemetry

Comment (wendigo) [reply docs/src/main/sphinx/release/release-458.md:9]: Or: Add OpenTelemetry tracing to JDBC driver

File: docs/src/main/sphinx/release/release-458.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+  support](file-system-configuration) with
+  `fs.native-azure.enabled`,`fs.native-gcs.enabled`, `fs.native-s3.enabled`, or
+  `fs.hadoop.enabled` in each catalog. Use the migration guides for [Azure
+  Storage](fs-legacy-azure-migration), [Google Cloud
+  Storage](fs-legacy-gcs-migration), and [S3](fs-legacy-s3-migration) to assist
+  if you have not switched from legacy support. ({issue}`23343`)
+* Add JMX monitoring to the [](/object-storage/file-system-s3). ({issue}`23177`)
+* Reduce the number of file system operations when reading from Delta Lake
+  tables. ({issue}`23329`)
+* Fix rare, long planning times when Hive metastore caching is enabled. ({issue}`23401`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-458.md:29]: fyi @@oskar-szwajkowski

**Pull request merged**